src/components/Tabs/useTabState.ts

```diff
@@ -9,12 +9,6 @@ export function useTabState(initialCurr: string, storeKey?: string): [string, (c
   // This ensures a nice SSR result, without waiting for the store to get set on the client!
   const [curr, setCurr] = useState<string>(initialCurr)
 
-  // If you're using a storeKey, set the shared store to an initial value
-  useEffect(() => {
-    if (storeKey) {
-      tabStore.setKey(storeKey, { curr: initialCurr })
-    }
-  }, [storeKey, initialCurr])
```

Comment:
@dwightjl , the issue is that this code resets the current value to the initial value when the next component renders. If the shared store needs to be set to the initial value, I can add "&& $tabStore[storeKey]?.curr == null" to the condition, and the code works fine (the store does not reset when the user sees the next component).

However, I don't understand why the initial value of the store needs to be set. I deleted this part of the code. Is there any reason why this code was there in the first place?

Reply:
Looks like this is not necessary. It might have been part of an effort to allow linking to specific tabs with a query parameter, but that capability was never added from what I can tell.